save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/WarnPlugin.kt

```diff
@@ -126,11 +127,17 @@ class WarnPlugin(
             .groupBy { if (it.line != null && it.column != null) it.line to it.column else null }
             .mapValues { (_, warning) -> warning.sortedBy { it.message } }
 
-        return TestResult(
-            paths,
-            checkResults(expectedWarnings, actualWarningsMap, warnPluginConfig),
-            DebugInfo(stdout, stderr, null)
-        )
+        return paths.map { path ->
+            TestResult(
+                listOf(path),
+                checkResults(
+                    expectedWarnings.filter { it.value.any { warning -> warning.fileName == path.name } },
+                    actualWarningsMap.filter { it.value.any { warning -> warning.fileName == path.name } },
+                    warnPluginConfig
+                ),
+                DebugInfo(stdout, stderr, null)
```

Comment:
stdout and stderr are still combined, right? We need to think how to remove redundant copies of this data

Reply:
fixed